resources/js/helpers/object.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Get a nested difference between two objects
+ * The values of source will be returned
+ *
+ * @param {Object} target
+ * @param {Object} source
+ *
+ * @returns Object
+ */
+export function objectDiff(target, source) {
+    return Object.fromEntries(
+        Object.entries(source)
+            .map(([key, val]) => {
+                if (!target || !(key in target)) {
+                    return [key, val]
+                }
+
+                if (target[key] === val || ((val === null || val === undefined) && (target[key] === null || target[key] === undefined))) {
+                    return null
+                }
```

Comment:
Should add an extra check here for `NaN` because `NaN === NaN` always returns false. The current setup would cause objects that have any value equal to NaN to always return that as a changed variable.